lib/chef/provider/windows_task.rb

```diff
@@ -214,6 +217,38 @@ def task_need_update?
         false
       end
 
+      def start_day_updated?
+        current_day = convert_system_date_format_into_ruby_date(@current_resource.start_day)
+        new_day = convert_system_date_format_into_ruby_date(new_resource.start_day)
+        current_day != new_day
+      end
+
+      def start_time_updated?
+        time = convert_system_date_format_into_ruby_date(@current_resource.start_time).strftime("%H:%M")
+        time != new_resource.start_time
+      end
+
+      def convert_system_date_format_into_ruby_date(date_in_string)
+        Chef::Log.debug "Looking system date format"
```

Comment:
Is this debug still useful? I think it can be deleted. Otherwise I'd change this to something like `"Converting '#{date_in_string}' to system date format"`